includes/class-bln-publisher.php

```diff
@@ -290,21 +290,44 @@ private function setup_client()
         $this->lightningClient = null;
         $this->lightningClientType = null;
 
-        if (!$this->lightningClient) {
-            try {
-                if (!empty($this->connection_options['lnd_address'])) {
+        if (!$this->lightningClient)
+        {
+            // Default to post author LNaddress
+            // if specified in user profile
+            if ( is_singular( array('post') ) )
+            {
+                global $post;
+                $address = get_user_meta( $post->post_author, '_lnp_ln_address', true );
+
+                if ( $address )
+                {
+                    $this->connection_options['lnaddress_address'] = $address;
```

Comment:
shouldn't we directly set the `lightningClient` here? 
```
$this->lightningClientType = 'lnaddress';
$this->lightningClient = new BLN_Publisher_LNAddress_Client($this->connection_options);
```

because otherwise we set the lnaddress connection option here, BUT if globally a `lnd_address` or `lnbits_apikey` etc. is set then this will still be preferred because below those are checked first. 

I'd think of something like: 

if it is a post page and if the post's author has a lightning address
then initialize and set the lightningClient with that address
else 
do what we do so far

Reply:
The reason for that was that I am not sure which client is enabled in options on the website by website owner.

Two clients that are using lnaddress on init are `BLN_Publisher_LND_Client` and `BLN_Publisher_LNAddress_Client`.
Which one do you think I should default in this case?

Reply:
I think we should do it like this: 

```php
if ( is_singular( array('post') ) )
            {
                global $post;
                $address = get_user_meta( $post->post_author, '_lnp_ln_address', true );

                if ( $address )
                {
                    $this->lightningClientType = 'lnaddress';
                    $this->lightningClient = new BLN_Publisher_LNAddress_Client($this->connection_options);
                    return; 
                }
}

continue as we do it right now

```


so we only add the addinonal check which take preference. 


IF it is a post and IF the author of that post has a `_lnp_ln_address` configured. 
THEN we set the `lightningClient` to use the lightning address


Otherwise we do it as it is right now and it checks the default site wide settings.

Reply:
@bumi Ok, I'm just not sure do we also need to update `lnaddress_address` like this:

```php
$this->connection_options['lnaddress_address'] = $address;
$this->lightningClient = new BLN_Publisher_LNAddress_Client($this->connection_options);
```

Otherwise `lightningClient` would use address from options instead the one from author.

Reply:
yes! true, did not pay attention there. (or we initialize the BLN_Publisher_LNAddress_Client not with the $this->connection option, but with the 'lnaddress_address ' => $address

Reply:
I didn't want to change code in `BLN_Publisher_LNAddress_Client`.
That's why I updated only `lnaddress_address` property before class init, technically this would not break anything somewhere else, and I went with the flow here.

Reply:
@bumi got it, you are right I updated code as per your suggestion